packages/zkdb/src/storage-engine/memory.ts

```diff
@@ -0,0 +1,115 @@
+import { StorageEngineBase } from './base.js';
+
+/**
+ * Storage engine using IPFS as backend
+ * @note This is a very simple implementation of storage engine using IPFS as backend
+ */
+export class StorageEngineMemory extends StorageEngineBase<string, any, any> {
+  private innerData = new Map<string, Uint8Array>();
+
+  /**
+   * Create new instance of storage engine
+   * @param basePath string Base path of the storage engine
+   */
+  constructor(pathBase: string) {
+    super(pathBase);
+  }
+
+  /**
+   * Make new folder in root folder and return CID of created folder
+   * @note If folder name is empty create a new root folder
+   * @todo Sync metadata every time we create new folder
+   * @param foldername Folder name
+   * @returns
+   */
+  public async mkdir(foldername: string = ''): Promise<string | undefined> {
+    return `${this.pathBase}/${foldername}`;
+  }
+
+  /**
+   * Check stat of a path
+   * @param path
+   * @returns
+   */
+  public async check(_path: string = ''): Promise<{} | undefined> {
+    return {};
+  }
+
+  /**
+   * Is path is a file
+   * @param path
+   * @returns Promise<boolean>
+   */
+  public async isFile(path: string = ''): Promise<boolean> {
+    const filepath = `${this.pathBase}/${path}`;
```

Comment:
String conjugation produces inconsistent results. We should use path.join instead

Reply:
> String conjugation produces inconsistent results. We should use path.join instead: path.join(this.pathBase, filePath)

It could be needless for IPFS. At least, I encountered an error suggesting a file didn't exist, even though it does. `path.join` helped me to overcome the issue

Reply:
`path.join()` used os separator that mean for Windows it would be `\` so your path might look like this:
`/ipfs\root\{CID}`.

Reply:
Btw, memory is a mock engine for test.